GitUp/Application/AppDelegate.m

```diff
@@ -69,6 +69,38 @@ - (AboutWindowController *)aboutWindowController {
   return _aboutWindowController;
 }
 
+- (WelcomeWindowController *)welcomeWindowController {
+  if (!_welcomeWindowController) {
+    _welcomeWindowController = [[WelcomeWindowController alloc] init];
+    
+    __weak NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+    _welcomeWindowController.model.getUserDefaultsShouldShow = ^BOOL{
+      return [defaults boolForKey:kUserDefaultsKey_ShowWelcomeWindow];
+    };
+    
+    __weak NSDocumentController *documentController = NSDocumentController.sharedDocumentController;
+    _welcomeWindowController.model.getRecentDocuments = ^NSArray<NSURL *> * _Nonnull{
+      return documentController.recentDocumentURLs;
+    };
+    
+    __weak typeof(self) weakSelf = self;
+    _welcomeWindowController.model.configureItem = ^(NSMenuItem * _Nonnull item) {
+      item.target = weakSelf;
+      item.action = @selector(_openDocument:);
+    };
+    
+    _welcomeWindowController.model.openTwitter = ^{
+      [[NSWorkspace sharedWorkspace] openURL:[NSURL URLWithString:kURL_Twitter]];
```

Comment:
Let the window handle what it's buttons do. We're not injecting anything useful from the `AppDelegate` here

Reply:
@lucasderraugh How to share URLs for twitter and for issues? For example, `-viewIssues:` method used both in welcome window and in main menu. Is it good to just set URLs for welcome window? ( twitterURL and forums/issuesURL )

Reply:
Ah, well if it's also called from the menu item, just let the responder chain handle that. Set the action to be targeting the First Responder object in the xib and select `viewIssues:` from the list.